scalyr_agent/builtin_monitors/linux_process_metrics.py

```diff
@@ -177,11 +178,21 @@ def __init__(self, pid, monitor_id, logger, file_pattern):
         self._logger = logger
         self._metric_printer = MetricPrinter(logger, monitor_id)
 
-    def run_single_cycle(self):
-        """Runs a single cycle of the sample collection.
+    def run_single_cycle(self, collector=None):
+        """
+        Runs a single cycle of the sample collection.
 
         It should read the monitored file and extract all metrics.
+        :param collector: Optional - a dictionary to collect the metric values.
+                          The collector has key as Metric, and value as the actual observed scalar.
+                          eg: {
+                                ("app.net.bytes", "in"): 12222,
+                                ("app.net.bytes", "out"): 20,
+                              }
+                          Note: a new collector will be instantiated if None is passed in.
+        :return: None or the optional collector with collected metric values
```

Comment:
I don't think it ever return `None`...  I think it either returns the collector that was passed in or the one it created.